_extras/guide.md

```diff
@@ -3,293 +3,548 @@ layout: page
+~~~
+{: .bash}
+
+- Let's look at the log with `--oneline` flag
+~~~
+$ git log --oneline
+~~~
+{: .bash}
+
+- Switch back to `master` and look at the log. New commit not on `master`
+~~~
+$ git checkout master
+$ git log --oneline
+~~~
+{: .bash}
+
+- We realize we want to add a clean up plan to the `README.md` file. Open editor and add the following:
+~~~
+Cleanup plan:
+- Get rid of all caps in titles
+- Standardize dates
+- Standardize dimensions
+~~~
+{: .bash}
+
+- Save file, add, and commit changes:
+~~~
+$ git add README.md
+$ git commit -m "add cleanup plan"
+~~~
+{: .bash}
+
+- Let's view a graph version of the log
+~~~
+$ git log --graph --oneline --decorate --all
+~~~
+{: .bash}
+
+### Merging Branches
+- Intro and context (bringing changes back into `master`)
+
+- Sometimes branches live permanently
+
+- Title edits exist in `title-caps`, not in `master`. Need to merge `title-caps` into `master`. Checkout `master`.
+~~~
+$ git checkout master
+~~~
+{: .bash}
+
+- **New Command**: `git merge` to bring commits into `master`.
+~~~
+$ git merge title-caps
+~~~
+{: .bash}
+
+- If editor window pops up, it's OK! We need to provide a merge commit message
+
+- Look at recent commits on master
+~~~
+$ git log --oneline
+~~~
+{: .bash}
+
+- We can now safely delete the `title-caps` branch. To verify changes were merged, we can run `branch` with `--merged`
+  flag from `master`
+~~~
+$ git checkout master
+$ git branch --merged
+$ git branch -d title-caps
+~~~
+{: .bash}
+
+### Key Points
+* "`git branch` to view current branches"
+* "`git branch branch-name` to create a new branch"
+* "`git checkout branch-name` to switch to a branch"
+* "`git branch -m branch-name` to rename a branch"
+* "`git merge branch-name` to merge a branch"
+* "`git branch -d delete-me` to delete a branch"
+* "Checking out a branch changes the “active” files in your repository directory."
+* "Merging applies commits from one branch to another."
+* "Merge commits are used to glue two histories together."
+
+## [Remotes]({{ page.root }}/07-remotes/)
+### Section Overview:
+- "How do I share my changes with others on the web?"
+
+### Instructions:
+- Log in to Github
+
+- Create repository
+
+- Copy `https` url of the repo
+
+- Add the repository as the `origin` remote. `origin` is a convention.
+~~~
+$ git remote add origin https://github.com/your-username/cats-as-data.git
+$ git remote -v
+~~~
+{: .bash}
+
+- **New Command**: `git push` to push our local repository content to our Github repository. Remote version of `commit`
+~~~
+$ git push origin master
+~~~
+{: .bash}
+
+- **New Command**: `git pull` to pull changes from our Github repository to our local repository. Pulling now has no
+  effect.
+~~~
+$ git pull origin master
+~~~
+{: .bash}
+
+### Key Points
+* "A local Git repository can be connected to one or more remote repositories."
+* "Use the HTTPS protocol to connect to remote repositories until you have learned how to set up SSH."
```

Comment:
I think HTTPS is recommended over SSH anyway.

Reply:
Really? Interesting.

Reply:
At least by GitHub, maybe just because it's more reliable from behind proxies: https://stackoverflow.com/a/11041782